docs/pre-commit.md

```diff
@@ -0,0 +1,33 @@
+# Using pre-commit locally
+
+### Install and setup pre-commit
+- Install:
+  - Using pip: `pip install pre-commit`
+  - Using Homebrew: `brew install pre-commit`
+
+- Setup
+  - run `pre-commit install` to set up the git hook script
+
+### Things to keep in mind regarding the git authoring lifecycle (HEAD, Staged, Unstaged) and pre-commit hooks
+
+Running hooks on unstaged changes can lead to both false-positives and false-negatives during committing. pre-commit only runs on the staged contents of files by temporarily saving the contents of your files at commit time and stashing the unstaged changes while running hooks.
+
```

Comment:
If pre-commit only runs on staged contents (2nd sentence), then how would you be a position to run hooks on unstaged changes (1st sentence)?

Reply:
That is a direct quote from the docs. https://pre-commit.com/#pre-commit-during-commits I'm not sure how to answer that question

Reply:
There is no way to run hooks on unstaged changes. This is part of the pre-commit design philosophy and is very important: pre-commit has side effects and by only running on staged changes you can always see the difference between what you wrote and what pre-commit changed.